storage: add value separation cluster settings

jbowens opened the pull request:
Add new cluster settings for enabling and configuring value separation.

Close #147709.
Epic: CRDB-20379
Release note (ops change): Introduces new cluster settings for enabling the preview feature of value separation.

jbowens committed:
storage: add value separation cluster settings

Add new cluster settings for enabling and configuring value separation.

Close #147709.
Epic: CRDB-20379
Release note (ops change): Introduces new cluster settings for enabling
the preview feature of value separation.

RaduBerinde commented on pkg/storage/pebble.go:
```diff
@@ -348,6 +348,33 @@ var concurrentDownloadCompactions = settings.RegisterIntSetting(
+	valueSeparationEnabled = settings.RegisterBoolSetting(
+		settings.SystemVisible,
+		"storage.value_separation.enabled",
+		"(experimental) whether or not values may be separated into external blob files; "+
```
nit: I'd remove external as it can be misinterpreted

RaduBerinde reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/148535#-:-OT85GPDE8KG_290lBH5:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @jbowens and @xxmplus)

___
*[`pkg/storage/pebble.go` line 358 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/148535#-OT84kgPEGyRM0u4v3_R:-OT84kgPEGyRM0u4v3_S:b-g8osvj) ([raw file](https://github.com/cockroachdb/cockroach/blob/6595cdbf1266747e1a46386d049a62b3074040ed/pkg/storage/pebble.go#L358)):*
> ```Go
> 			"requires columnar blocks to be enabled",
> 		metamorphic.ConstantWithTestBool(
> 			"storage.value_separation.enabled", true), /* defaultValue */
> ```

[nit] move `defaultValue` next to `true`
___
*[`pkg/storage/pebble.go` line 365 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/148535#-OT84maB0nKUVQE49Cw1:-OT84maB0nKUVQE49Cw2:bnkl5nw) ([raw file](https://github.com/cockroachdb/cockroach/blob/6595cdbf1266747e1a46386d049a62b3074040ed/pkg/storage/pebble.go#L365)):*
> ```Go
> 		"the minimum size of a value that will be separated into a blob file",
> 		int64(metamorphic.ConstantWithTestRange("storage.value_separation.minimum_size",
> 			1<<10 /* 1 KiB */, 25, 1<<20 /* 1 MiB */)),
> ```

[nit] add "default value" after 1KiB so it's obvious
___
*[`pkg/storage/pebble.go` line 372 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/148535#-OT84yWR7UAVhqMam61u:-OT84yWR7UAVhqMam61v:b-ivsh9h) ([raw file](https://github.com/cockroachdb/cockroach/blob/6595cdbf1266747e1a46386d049a62b3074040ed/pkg/storage/pebble.go#L372)):*
> ```Go
> 		"storage.value_separation.max_reference_depth",
> 		"the max reference depth bounds the number of unique, overlapping blob files referenced within a sstable;"+
> 			" lower values improve scan performance but harm write amplification",
> ```

[nit] s/harm/increase
___
*[`pkg/storage/pebble.go` line 373 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/148535#-OT852CV8ZxOL3tuq5Fw:-OT852CV8ZxOL3tuq5Fx:bvixq5r) ([raw file](https://github.com/cockroachdb/cockroach/blob/6595cdbf1266747e1a46386d049a62b3074040ed/pkg/storage/pebble.go#L373)):*
> ```Go
> 		"the max reference depth bounds the number of unique, overlapping blob files referenced within a sstable;"+
> 			" lower values improve scan performance but harm write amplification",
> 		int64(metamorphic.ConstantWithTestRange("storage.value_separation.max_reference_depth", 10, 2, 20)),
> ```

[nit] add a `/* defaultValue */` after 10


<!-- Sent from Reviewable.io -->

jbowens committed:
storage: construct roachpb.Values appropriately in unit tests

Fix unit tests that were improperly constructing roachpb.Values by directly
setting roachpb.Value's RawBytes field without the appropriate encoding.

Epic: none
Release note: none

jbowens reviewed (commented):
TFTR!

Added a commit that fixes some roachpb.Values in pkg/storage. I'm going to wait to merge this until cockroachdb/pebble#4914 is in incase there are other tests across the Cockroach codebase that write invalid values.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/148535#-:-OT8kUDx1WUyTtOaqwLJ:b3p64z8)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @xxmplus)



<!-- Sent from Reviewable.io -->

jbowens reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/148535)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @RaduBerinde and @xxmplus)

___
*[`pkg/storage/pebble.go` line 358 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/148535#-OT84kgPEGyRM0u4v3_R:-OT8tTky0jpCZxqj8Hj9:bw59srs) ([raw file](https://github.com/cockroachdb/cockroach/blob/6595cdbf1266747e1a46386d049a62b3074040ed/pkg/storage/pebble.go#L358)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

[nit] move `defaultValue` next to `true`
</blockquote></details>

crlfmt won't allow that.


<!-- Sent from Reviewable.io -->
